File: src/ios/MeteorMods.m
#import "MeteorMods.h"

#import <Cordova/CDVAvailability.h>

@implementation MeteorMods

- (void)pluginInitialize {
}

- (void)changeBackendUrl:(CDVInvokedUrlCommand *)command {
	NSString* url = [command.arguments objectAtIndex:0];
	CDVPluginResult *result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:url];
	[self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
}

@end

File: src/ios/MeteorMods.swift
@objc(MeteorMods)
class MeteorMods : CDVPlugin {
    @objc func changeBackendUrl(_ command: CDVInvokedUrlCommand) {
        let url = command.arguments[0] as? String ?? ""
        let configuration = WebAppConfiguration()
        configuration.rootURL = URL(string: url)
        
        let pluginResult = CDVPluginResult(
            status: CDVCommandStatus_OK,
            messageAs: "Done."
        )

        self.commandDelegate?.send(
            pluginResult,
            callbackId: command.callbackId
        )
    }
}